### ISSUE TITLE
"Missing new line after {" is reported incorrectly in nested lambda

### ISSUE THREAD

**Issue by vast00** [detekt/detekt#7286]
## Expected Behavior
```
    fun <T> remember(
        calculation: () -> T
    ): T {
        return calculation()
    }
    
    remember {
        { flag: Boolean ->
            println(flag)
        }
    }
```
should not have any lint warning.

## Observed Behavior
```
    fun <T> remember(
        calculation: () -> T
    ): T {
        return calculation()
    }
    
    remember {
         // Report "Missing new line after {" now
        { flag: Boolean ->
            println(flag)
        }
    }
```
It seems that the lambda is not recognized as lambda.

## Context
It's a common pattern in android that remembering a lambda in composable functions like this:
```
remember(scope) {
            // Reporting here
            { action ->
                scope.launch {
                    action.invoke()
                }
            }
        }
```

## Your Environment
* Version of detekt used: detekt jetbrains plugin 2.4.1 (Based on detekt 1.23.6)

**Comment by vast00** [comment]
Not stale

**Comment by cortinico** [comment]
What's the rule you're reporting this bug for?
